### ISSUE TITLE
[BUG]: Illegal instruction (core dumped) 

### ISSUE THREAD

**Issue by aequorea** [modularml/mojo#952]
### Bug description

I got mojo to install, but when I typed "mojo" to start the repl, I got "Illegal instruction (core dumped)". I'm running a Lenovo FlexPad 4. It has a "Intel(R) Pentium(R) CPU 4405U @ 2.10GHz". Woohoo!

### Steps to reproduce

I typed "mojo" at the command prompt. Worked every time!

### System information

```shell
- What OS did you do install Mojo on ?
- Provide version information for Mojo by pasting the output of `mojo -v`
- Provide Modular CLI version by pasting the output of `modular -v`
 
I'm running: Linux Mint...
Kernel: 5.15.0-84-generic x86_64 bits: 64 compiler: gcc v: 11.4.0 Desktop: Cinnamon 5.8.4
    tk: GTK 3.24.33 wm: muffin dm: LightDM Distro: Linux Mint 21.2 Victoria base: Ubuntu 22.04 jammy
 
I'm running the 'Illegal instruction' version of mojo, but I just installed it today. Fri Sep 28, 2023.
 
modular -v tells me: modular 0.2.0 (355ea9c0)
 
My processor is: Vendor ID: GenuineIntel
  Model name: Intel(R) Pentium(R) CPU 4405U @ 2.10GHz
    CPU family: 6
    Model: 78
    Thread(s) per core: 2
    Core(s) per socket: 2
    Socket(s): 1
    Stepping: 3
 
Have fun!
```

**Comment by River707** [comment]
Hmm, that looks a bit weird. Could you confirm if this happens with the newest version of the `modular tool` and `mojo`?

**Comment by aequorea** [comment]
OK, I tried to install the latest 0.5.0 on my flex4. This time the install does a bunch of tests that fail. For example:
[mojo] Testing: `mojo --help` [FAIL]
When I try:
mojo --version
I still get:
Illegal instruction (core dumped)

So yeah, its weird all right, but on the other hand it's kind of a new compiler. I ended up getting a 'new' laptop, a Thinkpad T480 and it works great there, so I'm not unable to run mojo. In fact I'm pleased to run it on my T480. That said I'm happy to keep testing on the old flex4 and it seems like it's the kind of thing that you would want to fix. Thanks!
